android/src/main/java/com/zmxv/RNSound/RNSoundModule.java

```diff
@@ -87,7 +88,13 @@ public synchronized boolean onError(MediaPlayer mp, int what, int extra) {
         return true;
       }
     });
-    player.prepareAsync();
+
+    try {
+      player.prepareAsync();
+    } catch (IllegalStateException ignored) {
+      // When loading files from a file, we useMediaPlayer.create, which actually
```

Comment:
... and detecting the state isn't possible. There's no 'isPrepared' or 'getState' on MediaPlayer :-(